### PR TITLE
Add provider side of certificate transfer interface

### PR DESCRIPTION
**Pull request by gatici**
# Description

- Send CA certificate by adding certificate transfer interface which is necessary to integrate with Traefik which requires knowing about Vault’s CA certificate.
- Enable metallb for integration tests. Traefik requires metallb and certifacate_transfer interface relation is tested with Traefik.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -123,6 +123,14 @@ vault operator raft list-peers
 juju integrate vault-k8s:metrics-endpoint prometheus-k8s:metrics-endpoint
 ```
 
+### Traefik
+
+Traefik requires to know about Vault’s CA certificate and the certificate-transfer is used to send VAult's CA certificate in the relation databag.
```
[the remark this answers is not in the document]
Comment (gatici) [reply README.md:128]: Done

File: .github/workflows/integration-test.yaml [hunk 2]
```diff
@@ -7,6 +7,16 @@ on:
...
+      enable-metallb:
+        description: Enable MetalLB
+        required: false
+        type: boolean
+        default: false
+      metallb-range:
+        description: MetalLB range
+        required: false
+        type: string
+        default: "10.0.0.2-10.0.0.10"
```
[the remark this answers is not in the document]
Comment (gatici) [reply .github/workflows/integration-test.yaml:19]: :)

File: src/charm.py [hunk 3]
```diff
@@ -824,6 +833,47 @@ def _set_pebble_plan(self) -> None:
...
+        send_ca_cert = CertificateTransferProvides(self, SEND_CA_CERT_RELATION_NAME)
+        if self._vault_ca_certificate_is_stored:
+            secret = self.model.get_secret(label=CA_CERTIFICATE_JUJU_SECRET_LABEL)
+            secret_content = secret.get_content()
+            ca = secret_content["certificate"]
+            if rel_id:
+                send_ca_cert.set_certificate("", ca, [], relation_id=rel_id)
+            else:
+                for relation in self.model.relations.get(SEND_CA_CERT_RELATION_NAME, []):
+                    send_ca_cert.set_certificate("", ca, [], relation_id=relation.id)
```
[the remark this answers is not in the document]
Comment (gatici) [reply src/charm.py]: Added. Done

File: src/charm.py [hunk 4]
```diff
@@ -824,6 +833,47 @@ def _set_pebble_plan(self) -> None:
...
+        Args:
+            rel_id: Relation id. If not given, update all relations.
+        """
+        send_ca_cert = CertificateTransferProvides(self, SEND_CA_CERT_RELATION_NAME)
+        if self._vault_ca_certificate_is_stored:
+            secret = self.model.get_secret(label=CA_CERTIFICATE_JUJU_SECRET_LABEL)
+            secret_content = secret.get_content()
+            ca = secret_content["certificate"]
+            if rel_id:
+                send_ca_cert.set_certificate("", ca, [], relation_id=rel_id)
```
[the remark this answers is not in the document]
Comment (gatici) [reply src/charm.py]: Done

**Review by patriciareinoso — APPROVED** [review]

**Review by ghislainbourgeois — APPROVED** [review]

**Pull request merged**